src/services/WorkspaceVariablesApi.ts

```diff
@@ -21,9 +21,13 @@ export class WorkspaceVariablesApi extends WorkspaceApi implements IWorkspaceVar
     return mapper.map('VariableResponse', data, 'Variable')
   }
 
-  public async getVariableByName(variableName: string): Promise<Variable> {
-    const { data } = await this.get<VariableResponse>(`/name/${variableName}`)
-    return mapper.map('VariableResponse', data, 'Variable')
+  public async getVariableByName(variableName: string): Promise<Variable | null> {
+    const { data } = await this.get<VariableResponse | null>(`/name/${variableName}`)
+    if (data) {
+      return mapper.map('VariableResponse', data, 'Variable')
+    }
+
+    return null
```

Comment:
Technically this will throw an error if the API can't find that variable but allowing this to return null allows us to create much more readable checks/error handling wherever this service is used